### PR TITLE
[CI-61] endpoint to save user session solution

### PR DESCRIPTION
**Pull request by socz3qqq**

### CODE REVIEW

File: backend/app/services/session.py [hunk 1]
```diff
@@ -68,3 +68,12 @@ async def get_session_ranking(db_session: AsyncSession, session_id: str) -> list
 
     rankings.sort(key=lambda x: x.score, reverse=True)
     return rankings
+
+async def asign_solution_to_user_session(db_session: AsyncSession, session_id: str, user_name: str, solution: QuizCheckIn) -> None:
```
Comment (rcybulski1122012) [inline backend/app/services/session.py]: I'd add some validation to make sure that such `Session` exists in the database

Comment (rcybulski1122012) [inline backend/app/services/session.py]:
```suggestion
async def assign_solution_to_user_session(db_session: AsyncSession, session_id: str, user_name: str, solution: QuizCheckIn) -> None:
```

File: backend/app/routers/session.py [hunk 2]
```diff
@@ -46,8 +48,21 @@ async def delete_session(
...
 
+
 @router.get("/{session_id}/ranking", response_model=list[UserSessionRankingOut])
 async def get_session_ranking(
     db_session: Annotated[AsyncSession, Depends(get_session)], session_id: str
 ) -> list[UserSessionRankingOut]:
     return await session_service.get_session_ranking(db_session, session_id)
+
+
+@router.post("/{session_id}/ranking/{user_name}/submit", response_model=QuizCheckOut)
```
Comment (rcybulski1122012) [inline backend/app/routers/session.py]: I'm not sure whether having a user name in the URL is a good idea 🤔

File: backend/app/routers/session.py [hunk 3]
```diff
@@ -46,8 +48,21 @@ async def delete_session(
...
+async def check_solution(
+    db_session: Annotated[AsyncSession, Depends(get_session)],
+    session_id: str,
+    user_name: str,
+    quiz_id: str,
```
Comment (rcybulski1122012) [inline backend/app/routers/session.py]: Having session_id we can easly retrieve the quiz_id, so we don't need it here.

Also it'll be treated as a query string parameter, which for me isn't the best way to pass parameters in this case (I'd use request's body)

File: backend/app/routers/session.py [hunk 4]
```diff
@@ -46,8 +48,21 @@ async def delete_session(
...
+    files_in: QuizCheckIn,
+) -> QuizCheckOut:
+    await session_service.asign_solution_to_user_session(db_session, session_id, user_name)
+    return await quiz_check_service.check_files(db_session, quiz_id, files_in)
```
Comment (rcybulski1122012) [inline backend/app/routers/session.py]: AFAIR Client didn't want to show the results to the student after submission. I'm wondering if we should limit the response 🤔 What do you think?

File: backend/app/routers/session.py (hunk 2 above)
Comment (socz3qqq) [reply backend/app/routers/session.py]: Renaming the endpoint as "/{session_id}/solution/submit" and placing the user_name in the request body will be a better option?

File: backend/app/routers/session.py (hunk 4 above)
Comment (socz3qqq) [reply backend/app/routers/session.py]: That's true, however we would need to create some mechanism for the admin, to allow students to see their score, so IMO for the time being we can stay with instant scoring.

File: backend/app/routers/session.py (hunk 2 above)
Comment (rcybulski1122012) [reply backend/app/routers/session.py]: IMO, yes

File: backend/app/routers/session.py (hunk 4 above)
Comment (rcybulski1122012) [reply backend/app/routers/session.py]: Right. So let's leave it as it is 😄

**Pull request merged**